xbout/options.py

```diff
@@ -0,0 +1,364 @@
+from collections import UserDict
+from pathlib import Path
+
+
+# These are imported to be used by 'evaluate=True' in Section.get().
+# Change the names to match those used by C++/BOUT++
+from numpy import (pi, sin, cos, tan, arccos as acos, arcsin as asin,
+                   arctan as atan, arctan2 as atan2, sinh, cosh, tanh,
+                   arcsinh as asinh, arccosh as acosh, arctanh as atanh,
+                   exp, log, log10, power as pow, sqrt, ceil, floor,
+                   round, abs)
+
+# TODO file reader
+# TODO substitution of keys within strings
+# TODO Sanitise some of the weirder things BOUT's option parser does:
+#  - Detect escaped arithmetic symbols (+-*/^), dot (.), or brackets ((){}[]) in option names
+#  - Detect escaping through backquotes in option names (`2ndvalue`)
+#  - Evaluate numletter as num*letter (and numbracket as num*bracket)
+#  - Substitute unicode representation of pi
+#  - Ensure option names don't contain ':' or '='
+#  - Check if all expressions are rounded to nearest integer?!
+# TODO ability to read from/write to nested dictionary
+
+
+SECTION_DELIM = ':'
+COMMENT_DELIM = ['#', ';']
```

Comment:
BOUT++ has dropped support for `;` as a comment character, I doubt you'll come across an input file that uses them